src/test/java/com/puppycrawl/tools/checkstyle/internal/CommitValidationTest.java

```diff
@@ -206,7 +223,11 @@ private static int validateCommitMessage(String commitMessage) {
         final String trimRight = commitMessage.replaceAll("[\\r\\n]+$", "");
         final int result;
 
-        if (!ACCEPTED_COMMIT_MESSAGE_PATTERN.matcher(message).matches()) {
+        if (message.matches(REVERT_COMMIT_MESSAGE_REGEX_PATTERN)) {
+            // revert commits are excluded from validation
+            result = 0;
+        }
+        else if (!ACCEPTED_COMMIT_MESSAGE_PATTERN.matcher(message).matches()) {
```

Comment:
As commented in the original issue https://github.com/checkstyle/checkstyle/issues/12766#issuecomment-1445427789,
revert commits violate two of the commit message rules. It would be easier to exclude them like this.